src/language-css/printer-postcss.js

```diff
@@ -493,77 +491,155 @@ function genericPrint(path, options, print) {
 
       return node.value;
     }
-    // postcss-values-parser
-    case "value-value":
-    case "value-root":
-      return print("group");
 
-    case "value-comment":
-      return options.originalText.slice(locStart(node), locEnd(node));
-
-    case "value-comma_group":
-      return printCommaSeparatedValueGroup(path, options, print);
+    // postcss-values-parser
+    // case "value":
+    // case "root":
+    //   return print("group");
+
+    // case "comment":
+    //   return options.originalText.slice(locStart(node), locEnd(node));
+
+    // case "comma_group":
+    //   return printCommaSeparatedValueGroup(path, options, print);
+
+    // case "paren_group":
+    //   return printParenthesizedValueGroup(path, options, print);
+
+    case "color":
+      return node.value.toString();
+
+    case "function-call":
+      return node.toString();
+
+    // case "func":
+    //   return [
+    //     node.value,
+    //     insideAtRuleNode(path, "supports") && isMediaAndSupportsKeywords(node)
+    //       ? " "
+    //       : "",
+    //     print("group"),
+    //   ];
+
+    case "list": {
+      // TODO: Every node needs a `type`
+      node.type = "list";
```

Comment:
Prettier throws an error if any node is missing a `type` -- should sass-parser always set `type` internally?

Reply:
Does this mean that Prettier demands that every parser for every language conforms to a certain interface? That seems needlessly onerous.

The reason we don't set `type` for value-level nodes is that `type` represents the PostCSS notion of a node's type, rather than the Sass parser's more detailed notion. Since PostCSS doesn't have a notion of the type of expression-level constructs, we _only_ provide `sassType`.

Honestly probably the easiest thing to do here would be to monkey-patch `Expression.prototype` to set a `type` property that just forwards to `sassType` if that's what PostCSS requires.

Reply:
I think we can tell Prettier to consistently use `sassType` for this parser 👍